### PR TITLE
[wip] refactor test-cmd startup to use util scripts

### PR DESCRIPTION
**Pull request by stevekuznetsov**
This is blocked on #10047 

@deads2k could you please review the first commit? only contentious change here

### CODE REVIEW

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -151,7 +151,20 @@ export OPENSHIFT_PROFILE="${WEB_PROFILE-}"
 # Specify the scheme and port for the listen address, but let the IP auto-discover. Set --public-master to localhost, for a stable link to the console.
 echo "[INFO] Create certificates for the OpenShift server to ${MASTER_CONFIG_DIR}"
 # find the same IP that openshift start will bind to.  This allows access from pods that have to talk back to master
-SERVER_HOSTNAME_LIST="${PUBLIC_MASTER_HOST},$(openshift start --print-ip),localhost"
+if [[ -z "${ALL_IP_ADDRESSES-}" ]]; then
+    ALL_IP_ADDRESSES="$(openshift start --print-ip)"
+    SERVER_HOSTNAME_LIST="${PUBLIC_MASTER_HOST},localhost,172.30.0.1"
+            SERVER_HOSTNAME_LIST="${SERVER_HOSTNAME_LIST},kubernetes.default.svc.cluster.local,kubernetes.default.svc,kubernetes.default,kubernetes"
```
Comment (deads2k) [inline hack/test-cmd.sh]: Why are all the internal names being added here.  were they being added before too and I just didn't see it?

Comment (stevekuznetsov) [inline hack/test-cmd.sh]: @deads2k I took the contents of `configure_os_server` from `hack/util.sh` and pasted it into `hack/test-cmd.sh` to see what the diff was in config writing between the utility func and our impl here -- the internal names are always added if you use `configure_os_server` and don't provide your own IPs

**Comment by deads2k** [comment]
squash down.  I have no objection as long as I get deconflicted ports.

**Comment by stevekuznetsov** [comment]
>  I get deconflicted ports

for etcd?

**Comment by deads2k** [comment]
> for etcd?

for etcd and API server.

**Comment by stevekuznetsov** [comment]
Yeah, this should be fine. I'll wait to squash until the blocker goes in

**Pull request closed without merging**